Highlight Logic Fixes

nirdoshyadav opened the pull request:
Juan, I have made couple of fixes in the existing alogorithm
- If multiple sentences are on the same div, highlight the full div
- Updated the Regex, removed the ' ' after the term so it will also highlight the terms having '.' after the term

nirdoshyadav committed:
- Fixed issues in the functionality to highlight multiple terms at the same page
- If multiple sentences are on the same div, highlight the full div
- Updated the Regex, removed the ' ' after the term so it will also highlight the terms having '.' after the term

MASmedios commented on web/pdf_find_controller.js:
```diff
@@ -142,59 +142,36 @@ var PDFFindController = (function PDFFindControllerClosure() {
-             ++regex.lastIndex;
+
+        //Nirdosh: regular expression to match full sentence
+        var regex=new RegExp("[^.]* "+query+"[^.]*\.","gi");
```
Use single quotes with strings

MASmedios commented on web/pdf_find_controller.js:
```diff
@@ -142,59 +142,36 @@ var PDFFindController = (function PDFFindControllerClosure() {
+        var regex=new RegExp("[^.]* "+query+"[^.]*\.","gi");
+        var match;
+        // console.log(query);
+        while ((match = regex.exec(pageContent)) != null) {
```
Compare with !== to compare value and type:
http://www.w3schools.com/js/js_comparisons.asp

MASmedios commented on web/pdf_find_controller.js:
```diff
@@ -276,7 +253,7 @@ var PDFFindController = (function PDFFindControllerClosure() {
       }
       //scroll match into view
-      if (PDFJS.multiple !== undefined) {
+      if (PDFJS.multiple == undefined) {
```
Compare with === 
To compare undefine type use typeof:
if (typeof PDFJS.multiple === 'undefined')

MASmedios commented on web/pdf_find_controller.js:
```diff
@@ -326,15 +303,13 @@ var PDFFindController = (function PDFFindControllerClosure() {
               if (PDFJS.multiple === undefined) {
```
Use typeof

MASmedios commented on web/pdf_find_controller.js:
```diff
@@ -326,15 +303,13 @@ var PDFFindController = (function PDFFindControllerClosure() {
-                        self.calcSearchMatch(pageIdx);
+                    for(var j=0;j<PDFJS.multiple.length;j++){
+                     //get the data for the current page
+                     if(PDFJS.multiple[j].page == pageIdx+1){
```
Use ===

MASmedios commented on web/text_layer_builder.js:
```diff
@@ -431,22 +431,50 @@ var TextLayerBuilder = (function TextLayerBuilderClosure() {
-        appendTextToDiv(divIdx, 0, begin.offset, className);
+        var div = textDivs[divIdx];
+        var endIndex=begin.offset;
+        var spans = div.getElementsByTagName("span");
```
Use single quote

MASmedios commented on web/text_layer_builder.js:
```diff
@@ -431,22 +431,50 @@ var TextLayerBuilder = (function TextLayerBuilderClosure() {
+        var endIndex=begin.offset;
+        var spans = div.getElementsByTagName("span");
+        var content=bidiTexts[divIdx].str;
+        if(spans!= 'undefined' && spans.length>0){
```
Use !== and typeof

MASmedios commented on web/text_layer_builder.js:
```diff
@@ -431,22 +431,50 @@ var TextLayerBuilder = (function TextLayerBuilderClosure() {
-          span.appendChild(node);
-          div.appendChild(span);
-          return;
+          var spans=div.getElementsByTagName("span");
```
Use single quote

MASmedios commented on web/text_layer_builder.js:
```diff
@@ -431,22 +431,50 @@ var TextLayerBuilder = (function TextLayerBuilderClosure() {
-          div.appendChild(span);
-          return;
+          var spans=div.getElementsByTagName("span");
+          if(spans!= 'undefined' && spans.length>0){
```
Use typeof and !==

MASmedios commented on web/text_layer_builder.js:
```diff
@@ -468,15 +496,16 @@ var TextLayerBuilder = (function TextLayerBuilderClosure() {
         var highlightSuffix = (isSelected ? 'selected' : '');
 
         if (PDFJS.multiple === undefined) {
```
Use typeof

nirdoshyadav committed:
- Using '===' operator in place of double operator '=='
- using typeof for comparing undefined
- Made one more small change in the logic -  reset the indexes to 0

MASmedios commented:
Hi Nirdosh,
the original search feature is broken. When i try to use it only search terms in the same page:
![image](https://cloud.githubusercontent.com/assets/609163/9910186/2fda2956-5c9c-11e5-83b6-9a263b72b39d.png)

Can you take a look on this?

nirdoshyadav commented:
I am looking into this. This is broken from earlier so you can merge this pull request. Once I find the fix, I will create a new pull request.

Thanks,
